src/components/alert/alert.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useMemo } from 'react'
+import clsx from 'clsx'
+import {
+  InformationCircleSolid,
+  CheckCircleSolid,
+  XCircleSolid,
+  ExclamationSolid,
+  XSolid,
+} from 'components/icons'
+import Button from 'components/button'
+
+export enum AlertType {
+  info = 'info',
+  success = 'success',
+  warning = 'warning',
+  error = 'error',
+}
+
+/**
+ * Alert properties
+ */
+export type AlertProps = {
+  /** Type of the alert which can be either `AlertType.error` or `AlertType.warning` or `AlertType.success` or `AlertType.info` */
+  type?: AlertType
+  /** Alert title */
+  title: React.ReactNode
+  /** Alert content */
+  content?: React.ReactNode
+  /** Alert icon */
+  icon?: JSX.Element | boolean
+  /** Alert action buttons */
+  actions?: JSX.Element
+  /** Making alert closable. It would render (X) button */
+  closable?: boolean
+  /** Function to be called on pressing the (X) button */
+  onClose?: (event: React.MouseEvent<HTMLButtonElement>) => void
+}
+
+/** Alert component to render `info`, `warning`, `success` and `error` messages */
+export function Alert({
+  type = AlertType.info,
+  title,
+  content,
+  icon,
+  actions,
+  closable = false,
+  onClose,
+}: AlertProps) {
+  // Function to return icon, based on user input and alert type
+  const renderIcon = useMemo(() => {
+    if (icon === false) {
+      return
+    }
+
+    if (icon) {
+      return icon
```

Comment:
Get the `baseColor`

```jsx
const baseColor = useMemo(() => {
  if (type === AlertType.info) {
    return 'blue'
  } else if (type === AlertType.success) {
    return 'green'
  } else if (type === AlertType.warning) {
    return 'yellow'
  } else if (type === AlertType.error) {
    return 'red'
  }
  return 'blue'
}, [type])
```

and after we can use the baseColor as 

```jsx
return cloneElement(icon, { className: `w-6 h-6 text-${baseColor}-500`})
```

This way we won't have to explicitly set **icon** `className` in the prop.

Reply:
If the styles are getting removed by `purgeCSS`, update the `tailwind.config.js` with

```js
purge: {
    content: ['./src/**/*.tsx', './src/**/*.ts'],

    // These options are passed through directly to PurgeCSS
    // so add all the classes that could be computed using the `baseColor`
    options: {
      whitelist: ['text-red-400', 'text-green-400', 'text-yellow-400', 'text-blue-400'],
    }
  },
```

Reply:
And I think we can reduce the icon size `w-5 h-5`. **tailwindui** uses that size and so is our `Button`